src/cf.data.pre

```diff
@@ -4868,6 +4868,28 @@ DOC_START
+
+	request_last_read: The latest observed event in a series of successful request read events. See request_first_read event for the definition of a successful request read event. If the entire request was received in a single socket read(2), then this event is the same as request_first_read event.
+
+	request_first_write: Squid starts delivering an HTTP or FTP request to an origin server or cache_peer. If Squid retries or re-forwards the request, then this event corresponds to the last such attempt. For a given forwarded request, this event is the first successful write event for the request. A successful request write event corresponds to the completion of a successful socket write(2) call. REQMOD adaptation of written bytes (if any), Squid-to-peer TCP connection establishment and Squid-peer TLS handshake (if any) happen earlier. Failed socket writes do not trigger this event (and usually result in request forwarding errors).
+
+	request_last_write: The latest observed event in a series of successful request write events. See request_first_write event for the definition of a successful request write event. If the entire request was delivered in a single socket write(2), then this event is the same as request_first_write event.
+
+	response_first_read: Squid starts working with an incoming HTTP or FTP response. If Squid retries or re-forwards the request, receiving multiple responses, then this event corresponds to the last such response. For a given response, this event is the first successful read event for the response. A successful response read event corresponds to the completion of a successful socket read(2) call (including a zero-size/EOF read). Squid-to-peer TCP connection establishment and Squid-peer TLS handshake (if any) happen earlier. RESPMOD adaptation of read bytes (if any) happens later. Failed socket reads do not trigger this event (and usually result in response processing errors).
+
+	response_last_read: The latest observed event in a series of successful response read events. See response_first_read event for the definition of a successful response read event. If the entire response was received in a single socket read(2), then this event is the same as response_first_read event.
+
+	response_first_write: Squid starts delivering an HTTP or FTP response to a client. This event is the first successful write event for the response. A successful response write event corresponds to the completion of a successful socket write(2) call. RESPMOD adaptation of written bytes (if any) happens earlier. Failed socket writes do not trigger this event (and usually result in transaction errors).
+
+	response_last_write: The latest observed event in a series of successful response write events. See response_first_write event for the definition of a successful response write event. If the entire response was delivered in a single socket write(2), then this event is the same as response_first_write event.
+
+
```

Comment:
The PR implementation differs from the proposed definitions (for request_first_read/request_last_read):

`request_first_read`
1) valid requests: Just after request parsing starts (parseOneRequest())
2) invalid requests (created by abortRequestParsing(): Same as (1)
3) fake requests: When the fake request is created (buildFakeRequest())

`request_last_read`
1) valid requests: request_first_read (for parsed complete requests) or just after it reads the last request data block from the client socket (for parsed incomplete requests)
2) invalid requests  (created by abortRequestParsing(): same as `request_first_read`
3) fake requests: When the fake request is created (buildFakeRequest())

`request_first_write`
Just after the first request data block is written to the peer socket (CommIoCbParams/IOCB callbacks)

`request_last_write`
Just after the last request data block is written to the peer socket (CommIoCbParams/IOCB callbacks)

`response_first_read`    
Just after the first response data block is read from the peer socket (CommIoCbParams/IOCB callbacks)

`response_last_read`    
Just after the last response data block is read from the peer socket (CommIoCbParams/IOCB callbacks)

`response_first_write`
Just after the first request data block is written to the client socket (CommIoCbParams/IOCB callbacks)

`response_last_write`
Just after the last request data block is written to the client socket (CommIoCbParams/IOCB callbacks)

----
We could simplify request_first_read/request_last_read calculation, ignoring parsing time and measuring them from ClientHttpRequest constructor. However, this approach would minimize request_first_read/request_last_read difference in some cases, concealing this useful information from the admin.